[Bug]: Can't install current develop branch artifact (clean install)

### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Doing a full build and using the install zip to install a new site is failing on the installer.

### Steps to reproduce?

1. Clone current develop branch
2. Run a full build (build.ps1)
3. Use nvQuickSite or a manual process to put the files in the hosting folder and create an empty database
4. Visit the site

### Current Behavior

The install fails

### Expected Behavior

The install should work

### Relevant log output

```shell
System.IO.FileNotFoundException: Could not load file or assembly 'System.Web.Http, Version=5.2.9.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies.

and

2023-08-08 13:09:31.748-04:00 [ERROR] DotNetNuke.Data.PetaPoco.PetaPocoHelper - [1] Error executing SQL: ;Exec dbo.GetDatabaseVersion
```


### Anything else?

_No response_

### Affected Versions

9.12.1 (alpha)

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/DNN Platform/Library/Common/Extensions/HttpContextDependencyInjectionExtensions.cs
+++ b/DNN Platform/Library/Common/Extensions/HttpContextDependencyInjectionExtensions.cs
@@ -4,6 +4,7 @@
 
 namespace DotNetNuke.Common.Extensions
 {
+    using System.Collections;
     using System.Web;
 
     using Microsoft.Extensions.DependencyInjection;
@@ -51,9 +52,10 @@ namespace DotNetNuke.Common.Extensions
         /// <returns>A service scope.</returns>
         public static IServiceScope GetScope(this HttpContextBase httpContext)
         {
-            if (httpContext.Items.Contains(typeof(IServiceScope)))
+            var scope = httpContext.Items.GetScope();
+            if (scope is not null || Globals.DependencyProvider is null)
             {
-                return httpContext.Items[typeof(IServiceScope)] as IServiceScope;
+                return scope;
             }
 
             var scopeLock = new object();
@@ -64,7 +66,6 @@ namespace DotNetNuke.Common.Extensions
                 return GetScope(httpContext);
             }
 
-            IServiceScope scope = null;
             httpContext.Items.Add(ScopeLockName, scopeLock);
             lock (httpContext.Items[ScopeLockName])
             {
@@ -87,9 +88,14 @@ namespace DotNetNuke.Common.Extensions
             return GetScope(httpContext);
         }
 
+        private static IServiceScope GetScope(this IDictionary httpContextItems)
+        {
+            return httpContextItems[typeof(IServiceScope)] as IServiceScope;
+        }
+
         private static void DisposeScope(HttpContextBase httpContext)
         {
-            httpContext.GetScope()?.Dispose();
+            httpContext.Items.GetScope()?.Dispose();
             httpContext.ClearScope();
         }
     }

